### ISSUE TITLE
The app crashed when install ADB

### ISSUE THREAD

**Issue by darouwan** [Simizfo/WSATools#24]
When I click the install button in the app, it shows "Downloading" for a while, then the app closes by itself. When I open the app again, it prompts me to install ADB again and crash happens again.

**Comment by laxmanrec** [comment]
Microsoft defender smart screen keeps not allowing the ADB to run and even I try to download ADB directly ,it still keep loading . Defender Also says can run the all.

**Comment by Simizfo** [comment]
Hi there. Thanks for reporting.

Can I ask you what country are you in? I've seen this happen with people in China.
Can you also please try download the [platform-tools](https://developer.android.com/studio/releases/platform-tools) package manually? Don't set it up, just try to download it. Can you? Do you need a VPN to access this file?

**Comment by darouwan** [comment]
> Hi there. Thanks for reporting.
> 
> Can I ask you what country are you in? I've seen this happen with people in China. Can you also please try download the [platform-tools](https://developer.android.com/studio/releases/platform-tools) package manually? Don't set it up, just try to download it. Can you? Do you need a VPN to access this file?

Yes, I am in China, and I can download the platform-tools via the proxy. May the crash be caused by the WSATools cannot work with the proxy?

**Comment by laxmanrec** [comment]
Hi

I am in Canada. The issue came after updating Magisk Google play from github, its keeps loading. Probably some compatibility issue. Nevermine I will wait for your next update



Regards
S.Lakshmanan
PH:+1(416)-720-8094

CONFIDENTIALITY NOTICE:
The contents of this email message and any attachments are intended solely for the addressee(s) and may contain confidential and/or privileged information and may be legally protected from disclosure. If you are not the intended recipient of this message or their agent, or if this message has been addressed to you in error, please immediately alert the sender by reply email and then delete this message and any attachments. If you are not the intended recipient, you are hereby notified that any use, dissemination, copying, or storage of this message or its attachments is strictly prohibited.

**Comment by Simizfo** [comment]
> Yes, I am in China, and I can download the platform-tools via the proxy. May the crash be caused by the WSATools cannot work with the proxy?

Yup. Will work to fix this. Thanks for reporting.

@laxmanrec I don't know how you got a mail to reply to this issue, but read issue #13, this is what you need to see.

**Comment by darouwan** [comment]
> > Yes, I am in China, and I can download the platform-tools via the proxy. May the crash be caused by the WSATools cannot work with the proxy?
> 
> Yup. Will work to fix this. Thanks for reporting.
> 
> @laxmanrec I don't know how you got a mail to reply to this issue, but read issue #13, this is what you need to see.

Also I see "This app can't run on your PC" message. May it also be one reason?

**Comment by Simizfo** [comment]
> Also I see "This app can't run on your PC" message. May it also be one reason?

This is Issue #21, I'm working on fixing this.

**Comment by QCXES** [comment]
You can leave me the adb file link. I am now in China and using proxy to access those things, and I can help to create an available one.

**Comment by LonelyInstantN** [comment]
Same crash problem on me. I'm in China too. But I have proxy and able to access ADB download site. I can download ADB manually, but apps crash after showing downloading. I think custom ADB directory is going to fix this. It seems the app won't use proxy, so it can't access the ADB download site.

**Comment by Simizfo** [comment]
I honestly have no idea on how to implement proxy usage in the app. Custom ADB Directory is almost ready. Just needs some testing. Will come in the next release! (#28)
